src/handler.ts

```diff
@@ -147,6 +148,28 @@ export default class DataHandler {
     return handledData;
   }
 
+  /**
+   * Returns false if there are any items in the data array which have first level child properties
+   * with defined values. Otherwise returns true. Returns true for empty data arrays.
+   * @param data the data array to be handled by operators
+   */
+  private static isEmptyData(data: any[]): boolean {
+    for (let i = 0; i < data.length; i += 1) {
+      const datum = data[i];
+
+      if (typeof datum !== 'object') {
+        return false;
+      }
+
+      // We don't use Object.values because it isn't supported by IE11
+      if (Object.keys(datum).some((key) => datum[key] !== undefined)) {
```

Comment:
~do we need to include `!== null` or use `!=`? unsure if setting something to `null` is possibly via data layer~
Seems like we are specifically expecting `null` to be a valid value, just curious why would we want `null`? Is it because `null` should always intentionally set vs. `undefined` isn't?

Reply:
This is a good question, and I'm open to suggestions for changes here. I approached this work by trying to make the `readOnLoad` behavior match the behavior for dynamic changes. Dynamic changes treated `null` as an empty object and more of an explicit set value rather than unset, so any `null` values would ultimately get suffixed with `_obj` and sent to `FS.event` as empty objects `{}`.

Maybe @van-fs has context for the original dynamic change behavior and whether this still makes sense.

Reply:
For  `handleData`, `null` is used to signal that the operator pipeline should stop processing (and not forward the data to the next operator).  An example of this is if we were to query and receive no result, which should stop processing.  Errors could also halt the pipeline by being caught and `null` returned by the operator.  So `null` has a certain semantics to continue or break.

I think that's why you're seeing `null` as a case for handling data but not the return value for readOnLoad is that when we read the data layer, we're always expecting an object (ideally a non-empty object).  It's worth confirming, but IIRC we have checks on when we monitor and read.  If the target (e.g. digitalData.something) is falsey, we should be backing off or not sending it to `handleEvent`.